src/main/kotlin/hs/flensburg/marlin/business/api/sensors/entity/UnitsWithLocationWithBoxesDTO.kt

```diff
@@ -0,0 +1,38 @@
+package hs.flensburg.marlin.business.api.sensors.entity
+
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class UnitsWithLocationWithBoxesDTO (
+    val units: MeasurementUnitsDTO,
+    val locationWithBoxes: List<LocationWithBoxesDTO>
+)
+
+fun mapToUnitsWithLocationWithBoxesDTO(rawLocations: List<LocationWithLatestMeasurementsDTO>): UnitsWithLocationWithBoxesDTO {
+    return UnitsWithLocationWithBoxesDTO(
+        units = MeasurementUnitsDTO(
+            getUnitFromRawLocations(rawLocations,"Temperature, water"),
+            getUnitFromRawLocations(rawLocations,"Wave Height"),
+            getUnitFromRawLocations(rawLocations,"Tide"),
+            getUnitFromRawLocations(rawLocations,"Standard deviation"),
+            getUnitFromRawLocations(rawLocations,"Battery, voltage"),
+            getUnitFromRawLocations(rawLocations,"Temperature, air"),
+            getUnitFromRawLocations(rawLocations,"Wind speed"),
+            getUnitFromRawLocations(rawLocations,"Wind direction"),
+            getUnitFromRawLocations(rawLocations,"Wind speed, gust"),
+            getUnitFromRawLocations(rawLocations,"Wind direction, gust"),
+            getUnitFromRawLocations(rawLocations,"Humidity, relative"),
+            getUnitFromRawLocations(rawLocations,"Station pressure")
+        ),
+        locationWithBoxes = rawLocations.map { it.mapToLocationWithBoxesDTO() }
+    )
+}
+
+fun getUnitFromRawLocations(rawLocations: List<LocationWithLatestMeasurementsDTO>, unitName: String): String {
+    return rawLocations.map { (_, measurements) ->
+        val unit = measurements.find { it.measurementType.name == unitName}?.measurementType?.unitSymbol
+        unit
```

Comment:
In Kotlin taucht `unit` häufig als Rückgabetyp von Lambdas auf. Ich fand das auf den ersten Blick etwas verwirrend, weil nicht sofort klar war, warum dort unit steht, es bezieht sich ja auf die Einheit.

An dieser Stelle musst du den Rückgabewert nicht extra in einer Variable speichern, da du nichts Weiteres damit machst, sondern ihn direkt zurückgeben willst. Ich würde das daher eher so schreiben:
```suggestion
        measurements.find { it.measurementType.name == unitName}?.measurementType?.unitSymbol
```